File: easy/173_without_repetitions.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main(int argc, const char * argv[]) {
    FILE *file = fopen(argv[1], "r");
    char line[1024];
    int i;
    char current;

    while (fgets(line, 1024, file)) {
        i = 0;
        current = line[0];

        while (i < strlen(line)) {
            if (line[i] != current) {
                printf("%c", current);
                current = line[i];
            }
            i++;
        }
        printf("%c", current);
    }

    return 0;
}

File: easy/173_without_repetitions.py
import sys

for line in open(sys.argv[1]):
    i = 0
    current = line[0]
    while i < len(line):
        if line[i] != current:
            sys.stdout.write(current)
            current = line[i]
        i += 1
    sys.stdout.write(current)

# The following re solution runs faster, but uses more memory and so gets a
# lower codeeval score than does the eyesore above.
#import sys
#import re
#
#for line in open(sys.argv[1]):
#    sys.stdout.write(re.sub(r'(.)\1+', r'\1', line))
